scripts/utils/docs-content-tools/handlers.js

```diff
@@ -0,0 +1,142 @@
+const glob = require('glob');
+const frontmatterGH = require('@github-docs/frontmatter');
+const fs = require('fs');
+
+/**
+ * Adds slashes to beginning and end of directory if missing
+ * @param {String} directory - The directory to add slashes to
+ * @returns {String} The directory with slashes added if necessary
+ */
+const formatDirectory = (directory) => {
+  !directory.startsWith('/') && (directory = `/${directory}`);
+  !directory.endsWith('/') && (directory = directory.concat('/'));
+  return directory;
+};
+
+/**
+ * Gets slug from path, relative to contents folder
+ * @param {String} filepath - The full filepath of the file
+ * @returns {String} The slug of the filepath
+ */
+const getSlugFromPath = (filepath) => {
+  return filepath.match(/content(.*?).mdx/)[1];
+};
+
+/**
+ * Reads a file and separates the content and frontmatter data
+ * @param {String} path - The full filepath of the file
+ * @returns {Object} Original path, slug from path, content, frontmatter data
+ */
+const separateData = async (path) => {
+  const file = fs.readFileSync(path, 'utf8');
+  const slug = getSlugFromPath(path);
+  const separateData = await frontmatterGH(file);
+  const { data: frontmatterData, content } = separateData;
+  if (!frontmatterData.redirects) {
+    frontmatterData.redirects = [];
+  }
+  return { path, content, frontmatterData, slug };
+};
+
+/**
+ * Joins the content and frontmatter data
+ * @param {Object} data - content, frontmatterData, path
+ * @returns {Object} Combined data and path
+ */
+const joinData = ({ content, frontmatterData, path }) => {
+  const data = frontmatterGH.stringify(content, frontmatterData, {
+    lineWidth: -1,
+  });
+
+  return { path, data };
+};
+
+/**
+ * Adds a redirect if the slug is not already in the redirects array
+ * @param {Object} data - slug, frontmatterData, rest
+ * @returns {Object} Edited frontmatterData, whether the redirect was added or skipped
+ */
+const addRedirect = ({ frontmatterData, slug, ...rest }) => {
+  if (frontmatterData.redirects.includes(slug)) {
+    return { frontmatterData, ...rest, skipped: slug };
+  }
+  frontmatterData.redirects.push(slug);
+  return {
+    frontmatterData,
+    ...rest,
+    added: slug,
+  };
+};
+
+/**
+ * Removes a redirect if the slug is in the redirects array
+ * @param {Object} data - slug, frontmatterData, rest
+ * @returns {Object} Edited frontmatterData, whether the redirect was removed or skipped
+ */
+const removeRedirect = ({ frontmatterData, slug, ...rest }) => {
+  if (frontmatterData.redirects.includes(slug)) {
+    frontmatterData.redirects = frontmatterData.redirects.filter(
+      (path) => path !== slug
+    );
```

Comment:
Since we're going to `filter` the redirects to remove the path anyway, I don't know if we need to bother checking if it exists. We can _probably_ remove the `if` statement.

Reply:
After chatting about this, it might make more sense to use `Array.prototype.map` instead of a `filter` here.

Reply:
Ok so the filter is actually to remove the slug from the redirects, then we also need to return an object with the data and a status result so i think it makes more sense to leave this